library/std/src/sys/unix/locks/pthread_condvar.rs

```diff
@@ -172,7 +172,7 @@ impl Condvar {
         let mut sys_now = libc::timeval { tv_sec: 0, tv_usec: 0 };
         let stable_now = Instant::now();
         let r = libc::gettimeofday(&mut sys_now, ptr::null_mut());
-        debug_assert_eq!(r, 0);
+        assert_eq!(r, 0, "unexpected error: {:?}", std::io::Error::last_os_error());
```

Comment:
POSIX says `gettimeofday()` isn't allowed to fail: https://pubs.opengroup.org/onlinepubs/9699919799/functions/gettimeofday.html.  I think it's okay to keep this as a debug assert.

Reply:
I make this because I read this https://man7.org/linux/man-pages/man2/gettimeofday.2.html that is the same linked [here](https://github.com/rust-lang/rust/pull/99624#discussion_r928030940)

What this is criteria of choice here? on what docs follow? I'm getting confused

Reply:
I think there's likely varying behaviors across platforms, but in general keeping the assert seems better to me.

Reply:
yes I agree @Mark-Simulacrum different os can return different result due custom conf